### PR TITLE
[Feature] Allow specifying all ports to use in advance

### PR DESCRIPTION
**Pull request by Ja1Zhou**

### CODE REVIEW

File: python/sglang/srt/utils.py [hunk 1]
```diff
@@ -98,6 +98,38 @@ def alloc_usable_network_port(num, used_list=()):
...
 
+def check_port(port):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        try:
+            s.bind(("", port))
+            return True
+        except socket.error:
+            return False
+
+def handle_port_init(port:int=None, additional_ports:list=None, tp_size:int=1):
```
Comment (merrymercy) [inline python/sglang/srt/utils.py]:
```suggestion
def handle_port_init(port: Optional[int] = None, additional_ports: Optional[List[int]] = None, tp_size: int = 1):
```

File: python/sglang/srt/utils.py [hunk 2]
```diff
@@ -98,6 +98,38 @@ def alloc_usable_network_port(num, used_list=()):
                 return port_list
     return None
 
+def check_port(port):
```
Comment (merrymercy) [inline python/sglang/srt/utils.py:102]:
```suggestion

def check_port(port):
```

Two empty lines between functions

File: python/sglang/srt/utils.py [hunk 3]
```diff
@@ -98,6 +98,38 @@ def alloc_usable_network_port(num, used_list=()):
...
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        try:
+            s.bind(("", port))
+            return True
+        except socket.error:
+            return False
```
Comment (merrymercy) [inline python/sglang/srt/utils.py:108]:
```suggestion
            return False

```

**Review by merrymercy — COMMENTED** [review]
LGTM! I only have a few minor notes about the style.

**Comment by Ja1Zhou** [comment]
Looks good to me!

**Pull request merged**